uxarray/grid.py

```diff
@@ -28,7 +27,7 @@ class Grid:
     >>> mesh.write("outfile.ug")
     """
 
-    def __init__(self, *args, **kwargs):
+    def __init__(self, data_arg, **kwargs):
```

Comment:
Th  dev team had thought there would be need for new args in the future, so they did this as `*args`. @rajeeja  @anissa111 any thoughts?

Reply:
I prefer an issue or a roadmap with that instead of "dead code parts or TODOs." It is easy to get lost and never address them. With that said, if I can revert that if you prefer it.

Reply:
That sounds good to me. FYI: Since the class constructor was being directly called by the user until this PR, the changing its signature would lead to broken calls on the user end, but now since we are planning the user to call the `open_dataset()` instead, this approach would work. Thanks!

Reply:
I renamed it to dataset everywhere for consistency. We are now passing it as an arg instead of a class attribute, and we close it after the `__from_ds__()` call just in case. Hope that is alright for now.